### PR TITLE
Added Golang-based operator website documentation

### LINKED ISSUE #2690: Organize golang docs  (website)
The scope of this issue is all files in website/content/en/docs/golang/

- [ ] add hugo frontmatter to all files
- [ ] subsection top-level should include (minimum):
  - [ ] Installation
  - [ ] Quickstart
  - [ ] Reference
- [ ] Place each file in relevant section
- [ ] update external links
- [ ] update section internal links. (Changes should only occur in content/en/docs/golang/, cross-section links will be updated later)

### PR DESCRIPTION
**Pull request by varshaprasad96**
**Description of the change:**
Organize Golang section for documentation website

Closes #2690

### CODE REVIEW

File: website/content/en/docs/golang/installation.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+local Kubernetes cluster and [quay.io][quay-link] for the public registry.
+
+[git_tool]:https://git-scm.com/downloads
+[go_tool]:https://golang.org/dl/
+[docker_tool]:https://docs.docker.com/install/
+[mercurial_tool]:https://www.mercurial-scm.org/downloads
+[kubectl_tool]:https://kubernetes.io/docs/tasks/tools/install-kubectl/
+[install-guide]: /docs/install-operator-sdk
+[minikube-tool]:https://github.com/kubernetes/minikube#installation
+[quay-link]:https://quay.io
```
Comment (jmrodri) [inline website/content/en/docs/golang/installation.md:28]: :+1: verified these links work

File: website/content/en/docs/golang/olm-catalog/_index.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+title: OLM - Caltalog
```
Comment (jmrodri) [inline website/content/en/docs/golang/olm-catalog/_index.md:2]: Typo: should be Catalog

File: website/content/en/docs/golang/metrics/operator-sdk-monitoring.md [hunk 3]
```diff
@@ -1,4 +1,8 @@
-#  Operator SDK Monitoring with Prometheus
+---
+title: Operator-SDK monitoring with Prometheus
```
Comment (jmrodri) [inline website/content/en/docs/golang/metrics/operator-sdk-monitoring.md:2]: I think this should be the proper name: "Operator SDK" with a space.

File: website/content/en/docs/golang/olm-catalog/bundle-cli.md [hunk 4]
```diff
@@ -1,4 +1,8 @@
-# Operator Bundle Tooling in `operator-sdk`
+---
+title: Operator Bundle Tooling in Operator-SDK
```
Comment (jmrodri) [inline website/content/en/docs/golang/olm-catalog/bundle-cli.md:2]: Use the proper name here as well: "Operator SDK"

File: website/content/en/docs/golang/metrics/service-monitor.md [hunk 5]
```diff
@@ -1,6 +1,10 @@
-# Using the ServiceMonitor prometheus-operator CRD with Operator SDK
+---
+title: Using the ServiceMonitor prometheus-operator CRD with Operator SDK
+linkTitle: Service-monitor
+weight: 1
+---
 
-[prometheus-operator][prom-operator] is an operator that creates, configures, and manages Prometheus clusters atop Kubernetes.
+[Prometheus-operator][prom-operator] is an operator that creates, configures, and manages Prometheus clusters atop Kubernetes.
```
Comment (estroz) [inline website/content/en/docs/golang/metrics/service-monitor.md:7]:
```suggestion
[`prometheus-operator`][prom-operator] is an operator that creates, configures, and manages Prometheus clusters atop Kubernetes.
```

File: website/content/en/docs/project_layout.md [hunk 6]
```diff
@@ -1,4 +1,8 @@
-# Project Scaffolding Layout for Operator SDK
+---
+title: Project Scaffolding Layout for Operator SDK
```
Comment (estroz) [inline website/content/en/docs/project_layout.md:2]: This should go under docs/golang

File: website/content/en/docs/golang/references/event-filtering.md [hunk 7]
```diff
@@ -1,4 +1,8 @@
-# Using Predicates for Event Filtering with Operator SDK
+---
+title: Using Predicates for Event Filtering with Operator SDK
+linkTitle: Event-filtering
```
Comment (joelanford) [inline website/content/en/docs/golang/references/event-filtering.md:3]:
```suggestion
linkTitle: Event Filtering
```

File: website/content/en/docs/golang/references/client.md [hunk 8]
```diff
@@ -1,4 +1,8 @@
-# Using the Controller Runtime Client API with Operator SDK
+---
```
Comment (joelanford) [inline website/content/en/docs/golang/references/client.md:1]: This whole doc is really about the controller-runtime client. Have we been keeping this up-to-date? Maybe we should pare this way down and provide links to controller-runtime's docs (or if they don't exist, contribute them upstream)?

This is non-blocking.

File: website/content/en/docs/golang/olm-catalog/olm-cli.md [hunk 9]
```diff
@@ -1,4 +1,8 @@
-# Running your Operator with OLM
+---
+title: Running your Operator with OLM
+linkTitle: OLM - cli
```
Comment (joelanford) [inline website/content/en/docs/golang/olm-catalog/olm-cli.md:3]:
```suggestion
linkTitle: Running with OLM
```

File: website/content/en/docs/golang/olm-catalog/generating-a-csv.md [hunk 10]
```diff
@@ -1,4 +1,8 @@
-# Generating a Cluster Service Version (CSV) with Operator SDK
+---
+title: Generating a Cluster Service Version (CSV) with Operator SDK
+linkTitle: Generating CSV
```
Comment (joelanford) [inline website/content/en/docs/golang/olm-catalog/generating-a-csv.md:3]:
```suggestion
linkTitle: Generating CSVs
```

File: website/content/en/docs/golang/olm-catalog/csv-annotations.md [hunk 11]
```diff
@@ -1,5 +1,8 @@
-# Code Annotations for Cluster Service Versions
-
+---
+title: Code Annotations for Cluster Service Versions
+linkTitle: CSV - Annotations
```
Comment (joelanford) [inline website/content/en/docs/golang/olm-catalog/csv-annotations.md:3]:
```suggestion
linkTitle: CSV Code Annotations
```

File: website/content/en/docs/golang/olm-catalog/bundle-cli.md [hunk 12]
```diff
@@ -1,4 +1,8 @@
-# Operator Bundle Tooling in `operator-sdk`
+---
+title: Operator Bundle Tooling in Operator SDK
+linkTitle: Bundle-cli
```
Comment (joelanford) [inline website/content/en/docs/golang/olm-catalog/bundle-cli.md:3]:
```suggestion
linkTitle: Working with Bundles
```

File: website/content/en/docs/golang/olm-catalog/_index.md [hunk 13]
```diff
@@ -0,0 +1,5 @@
+---
+title: OLM - Catalog
```
Comment (joelanford) [inline website/content/en/docs/golang/olm-catalog/_index.md:2]:
```suggestion
title: Integrating with Operator Lifecycle Manager
```

File: website/content/en/docs/golang/olm-catalog/_index.md [hunk 14]
```diff
@@ -0,0 +1,5 @@
...
+linkTitle: OLM - Catalog
```
Comment (joelanford) [inline website/content/en/docs/golang/olm-catalog/_index.md:3]:
```suggestion
linkTitle: Integrating with OLM
```

File: website/content/en/docs/golang/metrics/service-monitor.md [hunk 15]
```diff
@@ -1,6 +1,10 @@
-# Using the ServiceMonitor prometheus-operator CRD with Operator SDK
+---
+title: Using the ServiceMonitor prometheus-operator CRD with Operator SDK
```
Comment (joelanford) [inline website/content/en/docs/golang/metrics/service-monitor.md:2]:
```suggestion
title: Using the Prometheus Operator ServiceMonitor CRD with Operator SDK
```

File: website/content/en/docs/golang/metrics/service-monitor.md [hunk 16]
```diff
@@ -1,6 +1,10 @@
...
+linkTitle: Service-monitor
```
Comment (joelanford) [inline website/content/en/docs/golang/metrics/service-monitor.md:3]:
```suggestion
linkTitle: Using Prometheus ServiceMonitor
```

File: website/content/en/docs/golang/olm-catalog/_index.md [hunk 17]
```diff
@@ -0,0 +1,5 @@
+---
```
Comment (joelanford) [inline website/content/en/docs/golang/olm-catalog/_index.md:1]: Maybe also rename the directory to "olm-integration"?

File: website/content/en/docs/golang/references/client.md [hunk 18]
```diff
@@ -1,4 +1,8 @@
...
+title: Using the Controller Runtime Client API with Operator SDK
+linkTitle: Client API
```
Comment (joelanford) [inline website/content/en/docs/golang/references/client.md:3]:
```suggestion
linkTitle: Controller Runtime Client
```

File: website/content/en/docs/golang/unit-testing.md [hunk 19]
```diff
@@ -1,5 +1,8 @@
-# Unit testing with Operator SDK
-------------
+---
+title: Unit testing with Operator SDK
+linkTitle: Unit testing with Operator SDK
```
Comment (joelanford) [inline website/content/en/docs/golang/unit-testing.md:3]:
```suggestion
linkTitle: Unit Testing
```

File: website/content/en/docs/golang/references/logging.md [hunk 20]
```diff
@@ -1,4 +1,8 @@
-# Operator SDK Logging
+---
+title: Operator SDK Logging
+linkTitle: Operator SDK Logging
```
Comment (joelanford) [inline website/content/en/docs/golang/references/logging.md:3]:
```suggestion
linkTitle: Logging
```

File: website/content/en/docs/golang/references/client.md (hunk 8 above)
Comment (varshaprasad96) [reply website/content/en/docs/golang/references/client.md:1]: Yes, but I am not much sure on whether the doc is kept updated. Maybe @camilamacedo86 or @estroz could help in here.

**Comment by varshaprasad96** [comment]
@joelanford - I have made the suggested changes. Can you please have a look at it.

File: website/content/en/docs/golang/references/client.md (hunk 8 above)
Comment (asmacdo) [reply website/content/en/docs/golang/references/client.md:1]: IMO, this should be outside the scope of this PR. @joelanford could you file a separate issue for this?

File: website/content/en/docs/golang/olm-integration/_index.md [hunk 21]
```diff
@@ -0,0 +1,5 @@
+---
+title: Integrating with Operator Lifecycle Manager
+linkTitle: Integrating with OLM
+weight: 200
+---
```
Comment (asmacdo) [inline website/content/en/docs/golang/olm-integration/_index.md:5]: I *think* this section should be moved to the top level. I think this stuff isnt go-operator specific?

**Pull request merged**